lib/foreignNodeId.test.ts

```diff
@@ -0,0 +1,208 @@
+  it('should not work on non string types', async (): Promise<void> => {
+    const schema = ForeignNodeIdDirective.addValidationResolversToSchema(
+      makeExecutableSchema({
+        schemaDirectives: {
+          foreignNodeId: ForeignNodeIdDirective,
+        },
+        typeDefs: [
+          ...directiveTypeDefs,
+          gql`
+            input Input1 {
+              typeId: Int! @foreignNodeId(typename: "A")
+            }
+            type Query {
+              work(input: Input1!): Boolean
+            }
+          `,
+        ],
+      }),
+    );
+    const source = print(gql`
+      query MyQuery($input: Input1!) {
+        work(input: $input)
+      }
+    `);
+    const variables = {
+      input: {
+        typeId: 1,
+      },
+    };
+    const context = ForeignNodeIdDirective.createDirectiveContext({
+      fromNodeId,
+    });
+    const result = await graphql(schema, source, null, context, variables);
+    expect(result).toEqual({
+      data: { work: null },
+      errors: [
+        new ValidationError('foreignNodeId directive only works on strings'),
+      ],
+    });
+  });
+
+  it('typename does not match', async (): Promise<void> => {
+    const wrongName = 'wrong';
+    const typename = 'typename';
+    const schema = ForeignNodeIdDirective.addValidationResolversToSchema(
+      makeExecutableSchema({
+        schemaDirectives: {
+          foreignNodeId: ForeignNodeIdDirective,
+        },
+        typeDefs: [
+          ...directiveTypeDefs,
+          gql`
+            type Query {
+              work(arg: ID! @foreignNodeId(typename: "${typename}")): Boolean
+            }
+          `,
+        ],
+      }),
+    );
+    const source = print(gql`
+      query MyQuery($arg: ID!) {
+        work(arg: $arg)
+      }
+    `);
+    const variables = {
+      arg: toNodeId(wrongName, '1'),
+    };
+    const context = ForeignNodeIdDirective.createDirectiveContext({
+      fromNodeId,
+    });
+    const result = await graphql(schema, source, null, context, variables);
+    expect(result).toEqual({
+      data: { work: null },
+      errors: [
+        new ValidationError(
+          `Converted ID typename does not match. Received: ${wrongName} Expected: ${typename}`,
+        ),
+      ],
+    });
+  });
+
+  it('correctly convert types', async (): Promise<void> => {
+    const idsMap = [
+      { id: 'bbb', typeName: 'Type1' },
+      { id: 'id2', typeName: 'Type2' },
+      { id: 'id3', typeName: 'Type3' },
+      { id: 'id4', typeName: 'Type4' },
+      { id: 'aaaaa', typeName: 'Type5' },
+    ];
+    const sinonSpies: sinon.SinonSpy[] = [];
+    class TestDirective extends ForeignNodeIdDirective<ForeignNodeIdContext> {
+      public getValidationForArgs():
+        | ValidateFunction<ForeignNodeIdContext>
+        | undefined {
+        const func = super.getValidationForArgs();
+        if (func) {
+          const sinonSpy = sinon.spy(func);
+          sinonSpies.push(sinonSpy);
+          return sinonSpy;
```

Comment:
while sinon is okay, here you could just do with a `jest.fn()` outside of the class declaration.